solutions/instances/variables.tf

```diff
@@ -311,6 +311,16 @@ variable "cos_instance_tags" {
   default     = []
 }
 
+variable "retention_rule" {
+  description = "Defines the retention rule of the COS bucket."
+  type = object({
+    default   = optional(number, 90)
+    maximum   = optional(number, 365)
+    minimum   = optional(number, 90)
+    permanent = optional(bool, false)
+  })
+}
```

Comment:
Since the DA supports taking in an existing bucket, you can't make this a required input, so it needs to have a default value. Perhaps it should have a "best practise" default value that can be overridden? 

We will want to be careful here with tests - we don't want COS buckets getting created that cannot be destroyed.